components/segmented/segmented.spec.ts

```diff
@@ -172,21 +172,32 @@ describe('nz-segmented', () => {
     }));
   });
 
-  describe('icon-only', () => {
-    let fixture: ComponentFixture<NzSegmentedIconOnlyTestComponent>;
+  describe('DOM structure', () => {
+    let fixture: ComponentFixture<NzSegmentedDomStructureTestComponent>;
 
     beforeEach(() => {
-      fixture = TestBed.createComponent(NzSegmentedIconOnlyTestComponent);
+      fixture = TestBed.createComponent(NzSegmentedDomStructureTestComponent);
+      fixture.detectChanges();
+    });
+
+    it('should render an icon element and a text node with a wrapper element if the item is label-with-icon', () => {
```

Comment:
question: do we not need to inject the `provideNzIconsTesting` in the `TestBed.configureTestingModule()` providers options to avoid error on rendering icon ?

Reply:
Otherwise:  LGTM

Reply:
> question: do we not need to inject the `provideNzIconsTesting` in the `TestBed.configureTestingModule()` providers options to avoid error on rendering icon ?

In theory, any icon used in a test should use provideNzIconsTesting, which registers all icons. However, zorro registers some built-in icons by default (see /components/icon/icons.ts). If I only use these built-in icons, I can avoid using provideNzIconsTesting :)

Reply:
Ohhhh okkey i missed this information ! Thanks a lot 😀